sktime/forecasting/compose/_pipeline.py

```diff
@@ -135,17 +162,168 @@ def _fit(self, y, X=None, fh=None):
             Target time series to which to fit the forecaster.
         fh : int, list or np.array, optional (default=None)
             The forecasters horizon with the steps ahead to to predict.
-        X : pd.DataFrame, optional (default=None)
+        X : pd.DataFrame, required
             Exogenous variables are ignored
         Returns
         -------
         self : returns an instance of self.
         """
+        # Some transformers can not deal with X=None, therefore X is mandatory
+        self._set_y_X(y, X)
+        # copy to avoid transformation on original variable
+        Xt = _copy(self._X)
+
+        # If X is not given, just passthrough the data without transformation
+        if self._X is not None:
+            # transform X
+            for step_idx, name, transformer in self._iter_transformers():
+                t = clone(transformer)
+                Xt = t.fit_transform(Xt)
+                self.steps_[step_idx] = (name, t)
+
+        # fit forecaster
+        name, forecaster = self.steps[-1]
+        f = clone(forecaster)
+        f.fit(y, Xt, fh)
+        self.steps_[-1] = (name, f)
+
+        return self
 
+    def _predict(self, fh=None, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+        """Forecast time series at future horizon.
+
+        Parameters
+        ----------
+        fh : int, list, np.array or ForecastingHorizon
+            Forecasting horizon
+        X : pd.DataFrame, required
+            Exogenous time series
+        return_pred_int : bool, optional (default=False)
+            If True, returns prediction intervals for given alpha values.
+        alpha : float or list, optional (default=DEFAULT_ALPHA)
+
+        Returns
+        -------
+        y_pred : pd.Series
+            Point predictions
+        y_pred_int : pd.DataFrame - only if return_pred_int=True
+            Prediction intervals
+        """
+        forecaster = self.steps_[-1][1]
+        # copy to avoid transformation on original variable
+        Xt = _copy(X)
```

Comment:
I don't think we should copy the data here, I think the component transformers can take care of that, no? Otherwise we end up copying the data more often than we need to.